### DISCUSSION TITLE
i need help to conditionally display the createOptionForm based on permission

### DISCUSSION THREAD

**Discussion by Youssef-Elouasbi** [filamentphp/filament#13845] [Help] [answered]
### Package

Form builder

### Package Version

v3.2.61

### How can we help you?

i want to conditionally display the **createOptionForm** while ensuring the Select field remains visible when the user has this permission **"PaysCreate"** if not i want to display just the select without the option of creating new pays

i have this code 

```
Select::make('nationalite_id')
    ->label('Nationalité')
    ->relationship(name: 'pays', titleAttribute: 'nom')
    ->createOptionForm([
            Group::make()
                ->schema([
                    Section::make("Informations")
                        ->schema([
                            Hidden::make('cree_par')->default(auth()->user()->id),
                            Hidden::make('modifie_par')->default(auth()->user()->id),
                            Hidden::make('proprietaire')->default(auth()->user()->id),
                            TextInput::make('nom'),
                        ])->columns(3),
                ])->columnSpanFull(),
        ])
    ->searchable()
    ->preload();
```

**Comment by Youssef-Elouasbi** [comment, marked as answer]
this is the solution

```
Select::make('nationalite_id')
    ->label('Nationalité')
    ->relationship(name: 'pays', titleAttribute: 'nom')
    ->when(
        auth()->user()->can('PaysCreate'),
        fn ($select) => $select->createOptionForm([
            Group::make()
                ->schema([
                    Section::make("Informations")
                        ->schema([
                            Hidden::make('cree_par')->default(auth()->user()->id),
                            Hidden::make('modifie_par')->default(auth()->user()->id),
                            Hidden::make('proprietaire')->default(auth()->user()->id),
                            TextInput::make('nom'),
                        ])->columns(3),
                ])->columnSpanFull(),
        ])
    )
    ->searchable()
    ->preload();
```

**Reply by lukasinko** [reply]
@Youssef-Elouasbi Is this really secure solution ? I've tried this:

1. Invoke such `creation` operation with disabled `auth`, then i've captured `curl` command
2. Then i've enabled the auth with suggestion you've posted
3. Then i've tried to run the curl and it worked

Basically this `when(auth...)` only hides the button, but if one can guess (or somehow else) the url and command, he can execute it from eg. terminal. Would it probably make sense to protect the pays model directly ? Eg. via PayObserver#creating ?

**Reply by Youssef-Elouasbi** [reply]
Yes this code hide the buttton from the front end but in the same time i have a Policy for this model Pays .

for example in the policy i have this code 

```
 public function create(User $user): bool
    {
        if ($user->hasPermissionTo('PaysCreate')) {
            return true;
        }
        return false;
    }
```

**Reply by lukasinko** [reply]
@Youssef-Elouasbi Thank you. I guess the Pays model is also FilamentResource, right ? I had assumption, it is not a FilamentResource which would then need to have special treatement - like eg. PaysObserver#creating and checking the permission there. Do you agree ?
